near-sdk/src/private/abi.rs

```diff
@@ -0,0 +1,101 @@
+use std::collections::HashMap;
+
+use schemars::schema::{RootSchema, Schema};
+use serde::{Deserialize, Serialize};
+
+/// Current version of the ABI schema format.
+const ABI_SCHEMA_SEMVER: &str = "0.1.0";
+
+pub type TypeId = u32;
+
+/// Contract ABI.
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct AbiRoot {
+    /// Semver of the ABI schema format.
+    pub abi_schema_version: String,
+    /// Meta information about the contract.
+    pub metainfo: AbiMetainfo,
+    /// Core ABI information (functions and types).
+    pub abi: Abi,
+}
+
+impl AbiRoot {
+    pub fn new(abi: Abi) -> Self {
+        Self {
+            abi_schema_version: ABI_SCHEMA_SEMVER.to_string(),
+            metainfo: Default::default(),
+            abi,
+        }
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Default)]
+pub struct AbiMetainfo {
+    /// The name of the smart contract.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    /// The version of the smart contract.
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
Is there a reason these skip serializing if none but don't have `serde(default)` to allow deserializing without them?

Reply:
Simply forgot to update these fields :)